backend/tests/owasp/models/snapshot_test.py

```diff
@@ -0,0 +1,68 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+
+class TestSnapshotModel:
+    """Test cases for the Snapshot model."""
+
+    @pytest.mark.parametrize(
+        ("start_at", "end_at", "status", "expected_str"),
+        [
+            (
+                "2024-02-10 10:00:00",
+                "2024-02-10 12:00:00",
+                "pending",
+                "Snapshot 2024-02-10 10:00:00 to 2024-02-10 12:00:00 (pending)",
+            ),
+            (
+                "2024-02-11 14:00:00",
+                "2024-02-11 16:00:00",
+                "completed",
+                "Snapshot 2024-02-11 14:00:00 to 2024-02-11 16:00:00 (completed)",
+            ),
+        ],
+    )
+    def test_str_representation(self, start_at, end_at, status, expected_str):
+        snapshot = MagicMock(start_at=start_at, end_at=end_at, status=status)
+        snapshot.__str__.return_value = expected_str
+        assert str(snapshot) == expected_str
```

Comment:
This test doesn't make sense as you mock the `__str__` and compare it with the value you return. You need to actually test the code of Snapshot model